core/drivers/imx_uart.c

```diff
@@ -92,6 +92,9 @@ static void imx_uart_flush(struct serial_chip *chip)
 {
 	vaddr_t base = chip_to_base(chip);
 
+	if (!(read32(base + UCR1) & UCR1_UARTEN))
+		return;
+
```

Comment:
@jordanrh1 Just have a question, are you trying to resolve issue that Windows runtime disable the uart and then optee will panic when accessing uart?

Reply:
Yes, any subsequent boot stage can disable the UART including U-Boot, UEFI, or Windows.

Question to @jenswi-linaro  - what is the expected behavior if a subsequent boot stage disables the UART?

Reply:
The most useful behavior has to be to continue without prints. In many cases normal world isn't even aware secure world is using the uart for logging.